pages/d/[id].tsx

```diff
@@ -6,10 +6,15 @@ import Pusher from "pusher-js";
 import { M, mutators } from "../../frontend/mutators";
 import { randUserInfo } from "../../frontend/client-state";
 import { randomShape } from "../../frontend/shape";
+import { UndoManager } from "@rocicorp/undo";
 
 export default function Home() {
   const [rep, setRep] = useState<Replicache<M> | null>(null);
-
+  const [undoManager, setUndoManager] = useState<UndoManager | null>(null);
```

Comment:
I had undoManager as a ref but moved it back into state because of the onChange handler

Reply:
I don't understand why the onchange handler requires the undoManager to be state?